Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1284,21 +1281,24 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, unsigned int dstFo
   }
 
   // Invalidate all textures that overlap the range of our efb copy.
-  // Unless our efb copy has a weird stride, then we want avoid invalidating textures which
-  // we might be able to do a partial texture update on.
+  // Unless our efb copy has a weird stride, then we mark them to check for partial texture updates.
   // TODO: This also invalidates partial overlaps, which we currently don't have a better way
   //       of dealing with.
-  if (dstStride == bytes_per_row || !copy_to_vram)
+  bool invalidate_textures = dstStride == bytes_per_row || !copy_to_vram;
+  auto iter = FindOverlappingTextures(dstAddr, num_blocks_y * dstStride);
+  while (iter.first != iter.second)
   {
-    TexAddrCache::iterator iter = textures_by_address.begin();
-    while (iter != textures_by_address.end())
+    TCacheEntryBase* e = iter.first->second;
+    if (e->OverlapsMemoryRange(dstAddr, num_blocks_y * dstStride))
```

Comment:
As both are local variables, the compiler have already done so. But I've created a variable for this multiplication.